arrays/arrays.c

```diff
@@ -83,15 +101,26 @@ char *arr_read(Array *arr, int index) {
 void arr_insert(Array *arr, char *element, int index) {
 
   // Throw an error if the index is greater than the current count
-
+  if (index > arr->count) {
+    fprintf(stderr, "index is out of range.\n");
+    exit(1);
+  }
   // Resize the array if the number of elements is over capacity
+  if (arr->count == arr->capacity) {
+    resize_array(arr);
+  }
 
   // Move every element after the insert index to the right one position
+  for (int i = arr->count - 1; i >= index; i--) {
```

Comment:
Very nice, not many people would think to shift them over this way, with these specific bounding conditions.